File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.py
def solution(left, right):
    total = 0
    for num in range(left, right+1):
        cnt = 0
        for i in range(1, num+1):
            if num % i == 0:
                cnt += 1
        
        if cnt % 2 == 0:
            total += num
        else:
            total -= num
    
    return total

File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.java
class Solution {
    public int solution(int left, int right) {
        int answer = 0;
        int total = 0;
        // 약수의 개수 구하기
        for(int i=left; i<=right; i++) {
            for(int j=1; j<=i; j++) {
                if(i % j == 0)
                    answer++;
            }
            
            if (answer % 2 == 0) {
                total += i;
            } else {
                total -= i;
            }
            answer = 0;
        }

        return total;
    }
}